examples/experimental/h3-grid/src/h3-utils.js

```diff
@@ -0,0 +1,59 @@
+import {polyfill, getRes0Indexes, h3GetFaces, geoToH3} from 'h3-js';
+
+// Number of hexagons at resolution 10 in tile x:512 y:512 z:10
+// {west: 0, north: 0, east: 0.35156, south: -0.35156}
+const HEX_COUNT_ZOOM_10_RES_10 = 135439;
+// size multiplier when zoom increases by 1
+const ZOOM_FACTOR = 1 / 4;
+// size multiplier when resolution increases by 1
+// h3.numHexagons(n + 1) / h3.numHexagons(n)
+const RES_FACTOR = 7;
+
+export function getHexagonsInBoundingBox({west, north, east, south}, resolution) {
+  if (resolution === 0) {
+    return getRes0Indexes();
+  }
+  if (east - west > 180) {
+    // This is a known issue in h3-js: polyfill does not work correctly
+    // when longitude span is larger than 180 degrees.
+    return getHexagonsInBoundingBox({west, north, east: 0, south}, resolution).concat(
+      getHexagonsInBoundingBox({west: 0, north, east, south}, resolution)
+    );
+  }
+
+  return polyfill(
+    [[[west, north], [west, south], [east, south], [east, north], [west, north]]],
+    resolution,
+    true
+  );
+}
+
+export function getTileInfo(tile, resolution) {
+  if (!tile.centerHexagon) {
+    const {west, north, east, south} = tile.bbox;
+    const faces = [];
+
+    const NW = geoToH3(north, west, resolution);
+    faces.push(...h3GetFaces(NW));
+
+    const NE = geoToH3(north, east, resolution);
+    faces.push(...h3GetFaces(NE));
+
+    const SW = geoToH3(south, west, resolution);
+    faces.push(...h3GetFaces(SW));
+
+    const SE = geoToH3(south, east, resolution);
+    faces.push(...h3GetFaces(SE));
+
+    tile.hasMultipleFaces = new Set(faces).size > 1;
```

Comment:
Nit: use `const faces = new Set()` to begin with and avoid converting? Though adding multiple values isn't quite as easy, so maybe it's a wash.